lms/static/scripts/frontend_apps/components/GroupConfigSelector.js

```diff
@@ -118,18 +134,27 @@ export default function GroupConfigSelector({
         </LabeledCheckbox>
       </div>
       {useGroupSet && (
-        <div className="GroupConfigSelector__select">
-          {fetchError && (
+        <div>
+          {fetchError && isAuthorizationError(fetchError) && (
             // Currently all fetch errors are handled by attempting to re-authorize
             // and then re-fetch group sets.
-            <>
-              <p>Hypothesis needs your permission to fetch group sets</p>
-              <AuthButton
-                authURL={/** @type {string} */ (listGroupSetsAPI.authUrl)}
-                authToken={authToken}
-                onAuthComplete={fetchGroupSets}
-              />
-            </>
+            <AuthorizationModal
+              authURL={/** @type {string} */ (listGroupSetsAPI.authUrl)}
+              authToken={authToken}
+              onAuthComplete={fetchGroupSets}
+              onCancel={onErrorCancel}
+            >
+              <p>Hypothesis needs your permission to show group sets.</p>
+            </AuthorizationModal>
+          )}
+          {fetchError && !isAuthorizationError(fetchError) && (
+            <ErrorModal
+              cancelLabel="Cancel"
+              description="There was a problem fetching group sets"
```

Comment:
Like I mentioned in the PR's description, we can change this to anything we want...